src/block_ffm.rs

```diff
@@ -193,7 +211,13 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                             } else {
                                 for _z in 0..ffm_fields_count_as_usize {
                                     _mm_prefetch(mem::transmute::<&f32, &i8>(&ffm_weights.get_unchecked(feature_index + ffmk_as_usize)), _MM_HINT_T0);
-                                    for k in 0..ffmk_as_usize {
+                                    for k in (0..ffmk_end).step_by(STEP) {
+                                        let ffm_weights_simd = f32x4::from_slice(ffm_weights.get_unchecked(feature_index + k..feature_index + k + STEP));
```

Comment:
`k + STEP` tend to re-appear, to avoid extra additions perhaps this should be defined only once?